src/lasp_core.erl

```diff
@@ -912,6 +912,30 @@ write(Type, Value, Key, Store) ->
 increment_counter(Counter) ->
     Counter + 1.
 
+%% @private
+store_delta(Type, Counter, Delta, DeltaMap0) ->
+    MaxDeltaSlots = mochiglobal:get(delta_mode_max_slots, 10),
+    %% Check the space of the DeltaMap
+    case orddict:size(DeltaMap0) < MaxDeltaSlots of
+        true ->
+            %% Store a new delta.
+            orddict:store(Counter, Delta, DeltaMap0);
+        false ->
+            %% Find the minimum and 2nd minimum counters & those deltas.
+            [{MinCounter0, MinCounterDelta0}, {MinCounter1, MinCounterDelta1} | _Rest] =
+                orddict:to_list(DeltaMap0),
+            %% Merge them.
+            Merged = lasp_type:merge(Type,
+                                     MinCounterDelta0,
+                                     MinCounterDelta1),
+            %% Store the merged delta (minimum + 2nd minimum).
+            DeltaMap1 = orddict:store(MinCounter0, Merged, DeltaMap0),
+            %% Remove the 2nd minimum delta.
+            DeltaMap2 = orddict:erase(MinCounter1, DeltaMap1),
```

Comment:
I'm confused.  Doesn't this leave a gap?  For instance, if the bottom two elements in the orddict were 2, 3, it seems like you delete element 3 and update 2.  Wouldn't it be better to do the opposite to keep the numbering without gaps?  For instance, delete 2 after you merge 2 into 3?